src/plugins/data/public/actions/emit_apply_filter_trigger_action.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  ActionByType,
+  APPLY_FILTER_TRIGGER,
+  createAction,
+  UiActionsStart,
+} from '../../../../plugins/ui_actions/public';
+import { createFiltersFromRangeSelectAction } from './filters/create_filters_from_range_select';
+import { createFiltersFromValueClickAction } from './filters/create_filters_from_value_click';
+import type { Filter } from '../../common/es_query/filters';
+import type {
+  ChartActionContext,
+  RangeSelectContext,
+  ValueClickContext,
+} from '../../../embeddable/public';
+
+export const ACTION_EMIT_APPLY_FILTER_TRIGGER = 'ACTION_EMIT_APPLY_FILTER_TRIGGER';
```

Comment:
i don't think grouping the range select and value click is desired here. i would prefer to have two different actions (value_click_action, range_select_action), which are autoexecuted and emit a apply_filter action.